pkg/detectors/rootly/rootly.go

```diff
@@ -0,0 +1,77 @@
+package rootly
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	regexp "github.com/wasilibs/go-re2"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct{}
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	client = common.SaneHttpClient()
+
+	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
+	keyPat = regexp.MustCompile(`\brootly_[a-f0-9]{64}\b`)
+)
+
+// Keywords are used for efficiently pre-filtering chunks.
+// Use identifiers in the secret preferably, or the provider name.
+func (s Scanner) Keywords() []string {
+	return []string{"rootly"}
+}
+
+// FromData will find and optionally verify Rootly secrets in a given set of bytes.
+func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
+	dataStr := string(data)
+
+	matches := keyPat.FindAllStringSubmatch(dataStr, -1)
+
+	for _, match := range matches {
+		if len(match) != 1 {
+			continue
+		}
+		resMatch := strings.TrimSpace(match[0])
```

Comment:
Just curious, how did you generate this? This logic, present in some detectors, is old and doesn't conform to current best practices.

The "alchemy" detector seems to be the template that is good to follow.
https://github.com/trufflesecurity/trufflehog/blob/main/hack/docs/Adding_Detectors_external.md
https://github.com/trufflesecurity/trufflehog/blob/aa17b8eba486bbdf19bc77aedb3c56f9cb00e1ac/pkg/detectors/alchemy/alchemy.go#L4

Reply:
I actually took the reference from `alienvault` and tried to follow that, I will update this.

Reply:
@rgmz Can you check again?

Reply:
<img width="369" alt="Screenshot 2024-10-16 at 12 25 23 PM" src="https://github.com/user-attachments/assets/d1af7d6c-e34b-4605-bfb2-17a5874cc8d0">